src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -415,34 +415,50 @@ protected override int MakeSlot(BoundExpression node)
                         {
-                            var operand = conv.Operand;
-                            var operandType = operand.Type;
-                            var convertedType = conv.Type;
-                            if (operandType?.IsNullableType() == true &&
-                                convertedType?.IsNullableType() == false &&
-                                operandType.GetNullableUnderlyingType().Equals(convertedType, TypeCompareKind.AllIgnoreOptions))
-                            {
-                                // Explicit conversion of Nullable<T> to T is equivalent to Nullable<T>.Value.
-                                // For instance, in the following, when evaluating `((A)a).B` we need to recognize
-                                // the nullability of `(A)a` (not nullable) and the slot (the slot for `a.Value`).
-                                //   struct A { B? B; }
-                                //   struct B { }
-                                //   if (a?.B != null) _ = ((A)a).B.Value; // no warning
-                                int containingSlot = MakeSlot(operand);
-                                return containingSlot < 0 ? -1 : GetNullableOfTValueSlot(operandType, containingSlot);
-                            }
+                            case ConversionKind.ExplicitNullable:
+                                {
+                                    var operand = conv.Operand;
+                                    var operandType = operand.Type;
+                                    var convertedType = conv.Type;
+                                    if (operandType?.IsNullableType() == true &&
+                                        convertedType?.IsNullableType() == false &&
+                                        operandType.GetNullableUnderlyingType().Equals(convertedType, TypeCompareKind.AllIgnoreOptions))
+                                    {
+                                        // Explicit conversion of Nullable<T> to T is equivalent to Nullable<T>.Value.
+                                        // For instance, in the following, when evaluating `((A)a).B` we need to recognize
+                                        // the nullability of `(A)a` (not nullable) and the slot (the slot for `a.Value`).
+                                        //   struct A { B? B; }
+                                        //   struct B { }
+                                        //   if (a?.B != null) _ = ((A)a).B.Value; // no warning
+                                        int containingSlot = MakeSlot(operand);
+                                        return containingSlot < 0 ? -1 : GetNullableOfTValueSlot(operandType, containingSlot);
+                                    }
+                                }
+                                break;
+                            case ConversionKind.Identity:
+                            case ConversionKind.ImplicitTupleLiteral:
```

Comment:
>ImplicitTupleLiteral [](start = 48, length = 20)

What about ExplicitTupleLiteral. Is it covered?

Reply:
I've left other implicit and explicit conversions as a separate work item: [https://github.com/dotnet/roslyn/issues/32599](https://github.com/dotnet/roslyn/issues/32599).

---
In reply to: [249170170](https://github.com/dotnet/roslyn/pull/32612#discussion_r249170170) [](ancestors = 249170170)